tools/train.py

```diff
@@ -215,11 +218,23 @@ def main():
     model = build_detector(
         cfg.model, train_cfg=cfg.train_cfg, test_cfg=cfg.test_cfg)
 
+    datasets = [build_dataset(cfg.data.train)]
+
+    dataset_len_per_gpu = sum(len(dataset) for dataset in datasets)
+    if distributed:
+        dataset_len_per_gpu = dataset_len_per_gpu // get_dist_info()[1]
+    assert dataset_len_per_gpu > 0
     if cfg.data.samples_per_gpu == 'auto':
-        cfg.data.samples_per_gpu = determine_max_batch_size(cfg, distributed)
+        logger.info(f'Auto-selection of samples per gpu (batch size).')
+        cfg.data.samples_per_gpu = determine_max_batch_size(cfg, distributed, dataset_len_per_gpu)
+        logger.info(f'Auto selected batch size: {cfg.data.samples_per_gpu} {dataset_len_per_gpu}')
         cfg.dump(osp.join(cfg.work_dir, osp.basename(args.config)))
+    if dataset_len_per_gpu < cfg.data.samples_per_gpu:
```

Comment:
Just a note. If the dataset is that small, it may be a better choice to repeat it several times, rather than decrease the batch size. Though it'll require training schedule correction as well, making this a bit tricky. On the other hand training speed is  among the pros of extending the dataset, 'cause there is an overhead on starting a new epoch.

Reply:
To my mind it should not be the case in normal situation. Probably it is good enough workaround to see that training is going well with pre-defined configuration files from OTE and user-collected very small dataset. I would leave it as is, though your comments are reasonable for sure.